modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/compute/ClientComputeExecuteMapReduceRequest.java

```diff
@@ -0,0 +1,102 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler.requests.compute;
+
+import static org.apache.ignite.client.handler.requests.compute.ClientComputeExecuteRequest.unpackArgs;
+import static org.apache.ignite.client.handler.requests.compute.ClientComputeExecuteRequest.unpackDeploymentUnits;
+import static org.apache.ignite.client.handler.requests.compute.ClientComputeGetStatusRequest.packJobStatus;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.client.handler.NotificationSender;
+import org.apache.ignite.compute.DeploymentUnit;
+import org.apache.ignite.compute.JobStatus;
+import org.apache.ignite.compute.TaskExecution;
+import org.apache.ignite.internal.client.proto.ClientMessagePacker;
+import org.apache.ignite.internal.client.proto.ClientMessageUnpacker;
+import org.apache.ignite.internal.compute.IgniteComputeInternal;
+import org.jetbrains.annotations.Nullable;
+
+
+/**
+ * Compute MapReduce request.
+ */
+public class ClientComputeExecuteMapReduceRequest {
+
+    /**
+     * Processes the request.
+     *
+     * @param in Unpacker.
+     * @param out Packer.
+     * @param compute Compute.
+     * @param notificationSender Notification sender.
+     * @return Future.
+     */
+    public static CompletableFuture<Void> process(
+            ClientMessageUnpacker in,
+            ClientMessagePacker out,
+            IgniteComputeInternal compute,
+            NotificationSender notificationSender) {
+
+        List<DeploymentUnit> deploymentUnits = unpackDeploymentUnits(in);
+        String taskClassName = in.unpackString();
+        Object[] args = unpackArgs(in);
+
+        TaskExecution<Object> execution = compute.submitMapReduce(deploymentUnits, taskClassName, args);
+        sendTaskResult(execution, notificationSender);
+
+        var idsAsync = execution.idsAsync()
+                .handleAsync((ids, ex) -> {
+                    // empty ids in case of split exception to properly respond with task id and failed status
+                    return ex == null ? ids : Collections.<UUID>emptyList();
+                });
+        return execution.idAsync()
+                .thenAcceptBoth(idsAsync, (id, ids) -> {
+                    out.packUuidNullable(id);
```

Comment:
We can't return a null `id` to the client from the initial execution request, it does not make sense. The id should not be null here.